### PR TITLE
Add roadmap document

### PR DESCRIPTION
**Pull request by fgalan**
Fixes https://github.com/telefonicaid/iotagent-ul/issues/337

@dcalvoalonso could you have a look and provide LGTM if ok, pls? Thanks!

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  out of the scope of GE owners. Thus, the moment in which these features will be
+  finalized cannot be assured.
+  
+### Short term
+
+The following list of features are planned to be addressed in the short term, and incorporated in a next release of the product:
+- Docker image protects username and password using environment variables.
+- Removal of warnings and deprecation warnings.
+- Creation of tutorial (to be contributed by members of the Community beyond GE owners)
+
```
Comment (dcalvoalonso) [inline docs/roadmap.md]: Fix of bugs reported in GitHub repository, merge of open PRs

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+### Medium term
+
+The following list of features are planned to be addressed in the medium term, typically within the subsequent release(s) generated in the next 9 months after the next planned release:
+- Support to FIWARE complex data models.
+- Deployment instructions for high availability scenarios.
+- Incremental introduccion of ECMAScript6 syntax (previous analysis of which sub-set of interesting aspect we want to take)
```
Comment (dcalvoalonso) [inline docs/roadmap.md]: Deprecation of NGSIv1 could be also addressed in the medium term.
Implementation of lazy attributes.

File: docs/roadmap.md (hunk 1 above)
Comment (fgalan) [reply docs/roadmap.md]: Same as https://github.com/telefonicaid/iotagent-node-lib/pull/759#discussion_r257725079

NTC

File: docs/roadmap.md (hunk 2 above)
Comment (fgalan) [reply docs/roadmap.md]: Fixed in 356852e51af7fd9d65c9b50fd817003fb11b7bab

File: docs/roadmap.md (hunk 1 above)
Comment (fgalan) [reply docs/roadmap.md]: Same as https://github.com/telefonicaid/iotagent-node-lib/pull/759#discussion_r258005583

NTC

**Pull request merged**